views/js/qtiCreator/plugins/menu/preview.js

```diff
@@ -70,17 +84,18 @@ define([
                 title: __('Preview the item'),
                 text : __('Preview'),
                 cssClass: 'preview-trigger'
-            })).on('click', function previewHandler(e){
-                $(document).trigger('open-preview.qti-item');
-
-                e.preventDefault();
-
-                self.disable();
-
-                itemCreator.trigger('preview', itemCreator.getItem().data('uri'));
-
-                self.enable();
-            });
+            })).on('click', previewHandler.bind(this));
+
+            this.getAreaBroker()
+                .getItemPanelArea()
+                .on('dropped.gridEdit.insertable', function() {
+                    this.enable();
```

Comment:
Yes, this way is user-friendly, but we have error when user moves smth on canvas Preview enables, popup appears, user clicks 'Don't Save' and Test Preview opens with empty item and error in console
![image](https://user-images.githubusercontent.com/25976342/95959415-14b88c00-0e0b-11eb-85e7-fc6eb55dad43.png)
because getItem return empty item
![image](https://user-images.githubusercontent.com/25976342/95959590-55180a00-0e0b-11eb-8606-9a70ff41c115.png)
Better allow Preview only after save.

Reply:
Yes, similar to what Sergey saids, but if I do that we break current functionality. I mean, if add an interaction, save, then add another interaction and go to preview and click `don't save`, user should go to preview with the first interaction, so I need to handle `ifEmpty` and `ifSaved`.

I'm working on better approach, thanks!